aviatrix/resource_aviatrix_gateway.go

```diff
@@ -719,6 +754,14 @@ func resourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) error
 			d.Set("allocate_new_eip", true)
 		}
 
+		if gw.CloudType == 1 && (gw.EnableDesignatedGateway == "Yes" || gw.EnableDesignatedGateway == "yes") {
```

Comment:
Not sure if you need the gateway cloud type check, you can directly set it

Reply:
Yes, can directly set it.